src/main/java/com/google/sps/image/ImageSelection.java

```diff
@@ -48,47 +60,92 @@ public String generateSearchUrl() {
       }
     }
 
-    // Free to share and use commercially license
-    final String licenseFilter = "&qft=+filterui:license-L2_L3_L4";
-
-    final String bingQueryParam = "&qs=HS&form=QBIR&scope=images&sp=-1&pq=hap&sc=8-3&cvid=44CA4B129FEF4B93B6F764BD083213D3&first=1&scenario=ImageBasicHover";
+    final String bingQueryParam = "&form=IRFLTR&first=1&scenario=ImageBasicHover";
     final String safeSearchFilter = "&adlt=strict";
 
+    bingUrl = addFilter(bingUrl, copyrightFilter);
     bingUrl = addFilter(bingUrl, bingQueryParam);
-    bingUrl = addFilter(bingUrl, licenseFilter);
     bingUrl = addFilter(bingUrl, safeSearchFilter);
-
+    
     return bingUrl; 
   }
 
   /**
    * This is an endpoint. Call this function to get a relevant image.
    *
-   * @return  URL of the first image scraped from Bing Image Search.
+   * @param     analysationDepth Maximm number of different queries that the web crawler does. 
+   * @return    URL of the first image scraped from Bing Image Search.
    * @exception IOException if Bing doesn't map any image to the keywords.
    */
-  public String getBestImage() throws IOException {
-    List<String> imgSrc = new ArrayList<>();
+  public String getBestImage(int analysationDepth) throws IOException {
```

Comment:
This is a really long method, maybe try and split it in multiple methods for readability (if possible)

Reply:
Done.